howto/oom.md

```diff
@@ -0,0 +1,55 @@
+## Dealing with Out of Memory (OOM) errors:
+
+If you got this error while running a script
+
+```bash
+OutOfMemoryError: CUDA out of memory. Tried to allocate 2.22 GiB. GPU 0 has a total capacty of 79.15 GiB of which 228.38 MiB is free. Including non-PyTorch memory, this process
+has 78.93 GiB memory in use. Of the allocated memory 76.28 GiB is allocated by PyTorch, and 2.14 GiB is reserved by PyTorch but unallocated. If reserved but unallocated memory 
+is large try setting max_split_size_mb to avoid fragmentation.  See documentation for Memory Management and PYTORCH_CUDA_ALLOC_CONF
+```
+
+it means that your GPU memory size wasn't big enough for the model and script configuration.
+
+Here's a few things you can try:
+
+### Reduce the micro batch size
+
+Adjust the `micro_batch_size = ...` variable in the fine-tuning and pretraining scripts. This variable determines the number of samples loaded per iteration.
+
+A smaller value will simply load less samples simultaneously. The minimum value is 1.
+
+Experiment with different micro batch sizes to find a balance between memory consumption and computational efficiency. Smaller micro batch sizes consume less memory but may result in slower training convergence. Conversely, larger micro batch sizes require more memory but can accelerate training speed.
+
+### Try a different optimizer
+
+Our scripts use the [`AdamW` optimizer](https://pytorch.org/docs/main/generated/torch.optim.AdamW.html).
+It maintains 2 states for each trainable parameter of the model, meaning that the optimizer memory is double compared to
+an optimizer like [`SGD`](https://pytorch.org/docs/main/generated/torch.optim.SGD.html).
+You can try replacing it with your optimizer of choice that is lighter in memory requirements. keep in mind that different optimizers have distinct optimization behaviors, so it's essential to assess their impact on the training process and model performance.
+
+This suggestion is particularly relevant for pretraining, as the trainable parameters in the model represent a small
+subset of the total in the fine-tuning scripts.
+
+### Use lower precision
+
+Our scripts expose the `--precision` argument, this directly impacts the memory usage.
+
+Using true lower precision (`16-true`, `bf16-true`) reduces the memory usage by half compared to `32-true`, however,
+the model might start producing NaNs due to the limited range of representable values.
```

Comment:
Especially with `16-true`, or we're seeing a lot of this with `b16-true` as well?

Reply:
Both, at least for adapter